AzureTable/Microsoft.DataTransfer.AzureTable/Source/AzureTableSourceAdapter.cs

```diff
@@ -53,35 +59,53 @@ public AzureTableSourceAdapter(IAzureTableSourceAdapterInstanceConfiguration con
-            if (segmentDownloadTask == null)
-            {
-                MoveToNextSegment(null, cancellation);
-            }
+            TableQuerySegment<DynamicTableEntity> currentSegment = null;
+            RemoteLoggingClientProvider remoteLoggingClientProvider = new RemoteLoggingClientProvider();
+            IRemoteLogging remoteLogger = remoteLoggingClientProvider.GetRemoteLogger("tableapibulk");
 
-            var currentSegment = await segmentDownloadTask;
-
-            // Make sure current segment has data to read
-            while (currentEntityIndex >= currentSegment.Results.Count && currentSegment.ContinuationToken != null)
+            try
             {
-                MoveToNextSegment(currentSegment.ContinuationToken, cancellation);
+                if (segmentDownloadTask == null)
+                {
+                    MoveToNextSegment(null, cancellation);
+                }
+
                 currentSegment = await segmentDownloadTask;
-            }
 
-            if (currentEntityIndex >= currentSegment.Results.Count && currentSegment.ContinuationToken == null)
-            {
-                return null;
-            }
+                // Make sure current segment has data to read
+                while (currentEntityIndex >= currentSegment.Results.Count && currentSegment.ContinuationToken != null)
+                {
+                    MoveToNextSegment(currentSegment.ContinuationToken, cancellation);
+                    currentSegment = await segmentDownloadTask;
+                }
+
+                if (currentEntityIndex >= currentSegment.Results.Count && currentSegment.ContinuationToken == null)
+                {
+                    return null;
+                }
 
-            var entity = currentSegment.Results[currentEntityIndex++];
-            readOutput.DataItemId = entity.RowKey;
+                var entity = currentSegment.Results[currentEntityIndex++];
+                readOutput.DataItemId = entity.RowKey;
 
-            if (currentEntityIndex >= currentSegment.Results.Count && currentSegment.ContinuationToken != null)
+                if (currentEntityIndex >= currentSegment.Results.Count && currentSegment.ContinuationToken != null)
+                {
+                    // Start downloading next segment while current record is being processed
+                    MoveToNextSegment(currentSegment.ContinuationToken, cancellation);
+                }
+
+                return new DynamicTableEntityDataItem(AppendInternalProperties(entity));
+            }
+            catch(Exception excp)
             {
-                // Start downloading next segment while current record is being processed
-                MoveToNextSegment(currentSegment.ContinuationToken, cancellation);
+                // only enable remote logging for AzureTable -> Cosmos TableAPI
+                if (this.configuration.SinkContext.Equals("TableAPIBulk") && this.configuration.SourceContext.Equals("AzureTable"))
```

Comment:
We should avoid such checks for multiple reasons.
First of all, if I remember correctly - these names are configurable by the end-user through app.config, as they can register their own sinks and targets under same names.
Second, the whole concept of the tool is to support any-to-any. Otherwise you should just roll out your own simple tool that is tweaked to support one single point-to-point scenario, as it will be faster.

Reply:
good point. moved implementation to core, so N/A here.